diff --git a/api/config/v1/config.go b/api/config/v1/config.go
index 5df5284..6224fd9 100644
--- a/api/config/v1/config.go
+++ b/api/config/v1/config.go
@@ -22,8 +22,8 @@ import (
 	"os"

 	cli "github.com/urfave/cli/v2"
-	"k8s.io/klog/v2"
+	_ "k8s.io/klog/v2"

 	"sigs.k8s.io/yaml"
 )
@@ -83,17 +82,7 @@ func NewConfig(c *cli.Context, flags []cli.Flag) (*Config, error) {
 // DisableResourceNamingInConfig temporarily disable the resource renaming feature of the plugin.
 // This may be reenabled in a future release.
 func DisableResourceNamingInConfig(config *Config) {
-	// Disable resource renaming through config.Resource
-	if len(config.Resources.GPUs) > 0 || len(config.Resources.MIGs) > 0 {
-		klog.Warning("Customizing the 'resources' field is not yet supported in the config. Ignoring...")
-	}
-	config.Resources.GPUs = nil
-	config.Resources.MIGs = nil
-
-	// Disable renaming / device selection in Sharing.TimeSlicing.Resources
-	config.Sharing.TimeSlicing.disableResoureRenaming("timeSlicing")
-	// Disable renaming / device selection in Sharing.MPS.Resources
-	config.Sharing.MPS.disableResoureRenaming("mps")
+	// keep resource renaming intact; no-op
 }
 
 // parseConfig parses a config file as either YAML of JSON and unmarshals it into a Config struct.
